sdk/core/azure-core-http-vertx/src/main/java/com/azure/core/http/vertx/implementation/VertxRequestWriteSubscriber.java

```diff
@@ -134,7 +139,17 @@ private void onErrorInternal(Throwable throwable) {
 
     private void resetRequest(Throwable throwable) {
         subscription.cancel();
-        promise.fail(throwable);
+        if (!promise.tryFail(throwable)) {
+            // Seems the promise has already completed in some form.
+            // Attempt to associate this error with the existing failure.
+            Throwable cause = promise.future().cause();
+            if (cause != null) {
+                cause.addSuppressed(throwable);
```

Comment:
Log a debug message here so we don't lose the exception.

Reply:
This should be logged by the `HttpLoggingPolicy`.